lib/question.js

```diff
@@ -508,7 +500,13 @@ module.exports = {
             }
             if (ERR(err, callback)) return;
             debug('_gradeVariantWithClient()', 'success');
-            callback(null);
+            if (!data.broken && grading_job.grading_method == 'External') {
+                // We'll need to queue this job for grading once this
+                // transaction completes
+                callback(null, grading_job.id);
```

Comment:
The return value should be documented in the comments preceding the function, especially the fact that the returned `grading_job_id` is only non-`null` if we need to submit it for external grading.